### PR TITLE
Add pytest to Resource Mapping Tool requirements.txt

### PR DESCRIPTION
**Pull request by allisaurus**
When using a separate venv, was unable to run unit tests due to lack of `pytest`. 
Pulling in at least version `5.3.2` since that's what's specified in [o3de/python/requirements.txt](https://github.com/o3de/o3de/blob/development/python/requirements.txt#L214)

Also did an editorial pass of the README.

## Testing

requirements.txt validation
```
// before : tests don't run b/c no pytest
(.env) PS C:\Users\USER\source\o3de\Gems\AWSCore\Code\Tools\ResourceMappingTool> python -m pytest -vv .
C:\Users\USER\source\o3de\Gems\AWSCore\Code\Tools\ResourceMappingTool\.env\Scripts\python.exe: No module named pytest

// install new requirements.txt
(.env) PS C:\Users\USER\source\o3de\Gems\AWSCore\Code\Tools\ResourceMappingTool> pip install -r requirements.txt

Requirement already satisfied: PySide2>=5.15.2 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from -r requirements.txt (line 1)) (5.15.2)
Requirement already satisfied: boto3>=1.17.30 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from -r requirements.txt (line 2)) (1.20.21)
Collecting pytest>=5.3.2
  Using cached pytest-6.2.5-py3-none-any.whl (280 kB)
Requirement already satisfied: shiboken2==5.15.2 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from PySide2>=5.15.2->-r requirements.txt (line 1)) (5.15.2)
Requirement already satisfied: jmespath<1.0.0,>=0.7.1 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from boto3>=1.17.30->-r requirements.txt (line 2)) (0.10.0)
Requirement already satisfied: botocore<1.24.0,>=1.23.21 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from boto3>=1.17.30->-r requirements.txt (line 2)) (1.23.21)
Requirement already satisfied: s3transfer<0.6.0,>=0.5.0 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from boto3>=1.17.30->-r 
requirements.txt (line 2)) (0.5.0)
Collecting colorama
  Using cached colorama-0.4.4-py2.py3-none-any.whl (16 kB)
Collecting atomicwrites>=1.0
Collecting attrs>=19.2.0
  Using cached attrs-21.2.0-py2.py3-none-any.whl (53 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting packaging
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting py>=1.8.2
  Using cached py-1.11.0-py2.py3-none-any.whl (98 kB)
Collecting toml
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting iniconfig
  Using cached iniconfig-1.1.1-py2.py3-none-any.whl (5.0 kB)
Requirement already satisfied: urllib3<1.27,>=1.25.4 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from botocore<1.24.0,>=1.23.21->boto3>=1.17.30->-r requirements.txt (line 2)) (1.26.7)
Requirement already satisfied: python-dateutil<3.0.0,>=2.1 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from botocore<1.24.0,>=1.23.21->boto3>=1.17.30->-r requirements.txt (line 2)) (2.8.2)
Requirement already satisfied: six>=1.5 in c:\users\USER\source\o3de\gems\awscore\code\tools\resourcemappingtool\.env\lib\site-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.24.0,>=1.23.21->boto3>=1.17.30->-r requirements.txt (line 2)) (1.16.0)
Collecting pyparsing!=3.0.5,>=2.0.2
  Using cached pyparsing-3.0.6-py3-none-any.whl (97 kB)
Installing collected packages: pyparsing, toml, py, pluggy, packaging, iniconfig, colorama, attrs, atomicwrites, pytest
Successfully installed atomicwrites-1.4.0 attrs-21.2.0 colorama-0.4.4 iniconfig-1.1.1 packaging-21.3 pluggy-1.0.0 py-1.11.0 pyparsing-3.0.6 pytest-6.2.5 toml-0.10.2
WARNING: You are using pip version 21.2.3; however, version 21.3.1 is available.

// tests now run
(.env) PS C:\Users\USER\source\o3de\Gems\AWSCore\Code\Tools\ResourceMappingTool> python -m pytest -vv .
=============================================================================== test session starts ===============================================================================
platform win32 -- Python 3.9.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- C:\Users\USER\source\o3de\Gems\AWSCore\Code\Tools\ResourceMappingTool\.env\Scripts\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\USER\source\o3de, configfile: pytest.ini
collected 135 items
 
tests\unit\controller\test_error_controller.py::TestErrorController::test_setup_with_expected_behavior_connected PASSED                                                      [  0%] 
tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_back_button_switching_view_gets_invoked_and_resetting_page_with_expected_state PASSED [  1%]
tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_cfn_stacks_import_button_emit_signal_with_expected_resources_and_switch_to_expected_page PASSED [  2%]
tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_cfn_stacks_import_button_nothing_happened_when_selected_resources_are_empty PASSED [  2%]
... [some tests omitted]
tests\unit\utils\test_json_utils.py::TestJsonUtils::test_validate_resources_according_to_json_schema_return_expected_rows_when_resource_is_invalid PASSED                    [ 99%] 
tests\unit\utils\test_json_utils.py::TestJsonUtils::test_write_into_json_file_succeed PASSED                                                                                 [100%] 
 
=============================================================================== 135 passed in 1.96s ===============================================================================

```

Signed-off-by: allisaurus <34254888+allisaurus@users.noreply.github.com>

### CODE REVIEW

File: Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
    $ cmake -B <BUILD_FOLDER> -S . -G "Visual Studio 16 2019" -DLY_PROJECTS=<PROJECT_NAME>
    ```
 
-2. At this point, double check engine python environment gets setup under *<ENGINE_ROOT_PATH>/python/runtime* directory
+2. At this point, double check that the O3DE engine's python environment gets set up under *<ENGINE_ROOT_PATH>/python/runtime* directory
 
-3. Build project with **AWSCore.Editor** (or **AWSCore.ResourceMappintTool**, or **Editor**) target to generate required Qt binaries.
+3. Build the project with **AWSCore.Editor** (or **AWSCore.ResourceMappingTool**, or **Editor**) target to generate the required Qt binaries.
    ```
    $ cmake --build <BUILD_FOLDER> --target AWSCore.Editor --config <CONFIG> -j <NUM_JOBS>
```
Comment (spham-amzn) [inline Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md:27]: I know its not part of this change, but you can remove ```-J <NUM_JOBS>``` because it only works for Mac and Linux, on windows the argument is different ```/m```

**Review by spham-amzn — APPROVED** [review]

**Review by onecent1101 — APPROVED** [review]
Minor: LG, probably you can give an example of using pytest as you include it as requirements, something like
```
python -m pytest Gems\AWSCore\Code\Tools\ResourceMappingTool\tests\unit
```

**Comment by allisaurus** [comment]
@onecent1101 it kind of gets lost in the large code block, but at the bottom of the test out above, put you can see it's successfully running:
```
// tests now run
(.env) PS ...> python -m pytest -vv .
=============================================================================== test session starts ===============================================================================
platform win32 -- Python 3.9.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- C:\Users\USER\source\o3de\Gems\AWSCore\Code\Tools\ResourceMappingTool\.env\Scripts\python.exe
cachedir: .pytest_cache
rootdir: C:\Users\USER\source\o3de, configfile: pytest.ini
collected 135 items
 
tests\unit\controller\test_error_controller.py::TestErrorController::test_setup_with_expected_behavior_connected PASSED                                                      [  0%] 
tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_back_button_switching_view_gets_invoked_and_resetting_page_with_expected_state PASSED [  1%]
tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_cfn_stacks_import_button_emit_signal_with_expected_resources_and_switch_to_expected_page PASSED [  2%]
tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_cfn_stacks_import_button_nothing_happened_when_selected_resources_are_empty PASSED [  2%]
... [some tests omitted]
tests\unit\utils\test_json_utils.py::TestJsonUtils::test_validate_resources_according_to_json_schema_return_expected_rows_when_resource_is_invalid PASSED                    [ 99%] 
tests\unit\utils\test_json_utils.py::TestJsonUtils::test_write_into_json_file_succeed PASSED                                                                                 [100%] 
 
=============================================================================== 135 passed in 1.96s ===============================================================================
```

**Comment by onecent1101** [comment]
> @onecent1101 it kind of gets lost in the large code block, but at the bottom of the test output above, you can see it's successfully running:
> 
> ```
> // tests now run
> (.env) PS ...> python -m pytest -vv .
> =============================================================================== test session starts ===============================================================================
> platform win32 -- Python 3.9.7, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- C:\Users\USER\source\o3de\Gems\AWSCore\Code\Tools\ResourceMappingTool\.env\Scripts\python.exe
> cachedir: .pytest_cache
> rootdir: C:\Users\USER\source\o3de, configfile: pytest.ini
> collected 135 items
>  
> tests\unit\controller\test_error_controller.py::TestErrorController::test_setup_with_expected_behavior_connected PASSED                                                      [  0%] 
> tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_back_button_switching_view_gets_invoked_and_resetting_page_with_expected_state PASSED [  1%]
> tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_cfn_stacks_import_button_emit_signal_with_expected_resources_and_switch_to_expected_page PASSED [  2%]
> tests\unit\controller\test_import_resources_controller.py::TestImportResourcesController::test_page_cfn_stacks_import_button_nothing_happened_when_selected_resources_are_empty PASSED [  2%]
> ... [some tests omitted]
> tests\unit\utils\test_json_utils.py::TestJsonUtils::test_validate_resources_according_to_json_schema_return_expected_rows_when_resource_is_invalid PASSED                    [ 99%] 
> tests\unit\utils\test_json_utils.py::TestJsonUtils::test_write_into_json_file_succeed PASSED                                                                                 [100%] 
>  
> =============================================================================== 135 passed in 1.96s ===============================================================================
> ```

I mean add instruction in README, for how to use pytest

**Review by junbo75 — APPROVED** [review]
Thanks for making the updates, Allie.

File: Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md [hunk 2]
```diff
@@ -1,35 +1,40 @@
...
-Or follow **AWS CLI** configuration which can be reused by boto3 lib:
+Or follow **AWS CLI** configuration directions which can be reused by the boto3 lib:
  * Follow
    [Quick configuration with aws configure](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-config)
 
 ## Python Environment Setup Options
-### 1. Engine python environment (Including Editor)
-1. In order to use engine python environment, it requires to link Qt binaries for this tool.
+### 1. Use Engine python environment (Including Editor)
+1. In order to use the O3DE engine's python environment, this tool requires linking the Qt binaries.
```
Comment (lmbr-pip) [inline Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md:17]: nit: O3DE is the Open 3d engine, so it reads as the engine engine's python environment.

File: Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md [hunk 3]
```diff
@@ -1,35 +1,40 @@
...
 Follow cmake instructions to configure your project, for example:
    ```
    $ cmake -B <BUILD_FOLDER> -S . -G "Visual Studio 16 2019" -DLY_PROJECTS=<PROJECT_NAME>
    ```
 
-2. At this point, double check engine python environment gets setup under *<ENGINE_ROOT_PATH>/python/runtime* directory
+2. At this point, double check that the O3DE engine's python environment gets set up under *<ENGINE_ROOT_PATH>/python/runtime* directory
```
Comment (lmbr-pip) [inline Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md:23]: not: Open 3D engine engine's .... (similar point as above)

File: Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md [hunk 4]
```diff
@@ -49,10 +54,9 @@ Follow cmake instructions to configure your project, for example:
       $ python/python.sh Gems/AWSCore/Code/Tools/ResourceMappingTool/resource_mapping_tool.py --binaries_path <PATH_TO_BUILD_FOLDER>/bin/debug/AWSCoreEditorQtBin
       ```
       
-* Note - Editor is integrated with the same engine python environment to launch Resource Mapping Tool. If it is failed to launch the tool
-in Editor, please follow above steps to make sure expected scripts/binaries are present.
+* Note - the Editor is integrated with the same engine python environment to launch the Resource Mapping Tool. If it fails to launch the tool from the Editor, please follow above steps to make sure expected scripts and binaries are present.
```
Comment (lmbr-pip) [inline Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md:57]: Note: the Editor is integrated with the same engine python environment *used* to launch the resource mapping tool. If the tool fails to launch from Editor...

Feel like this paragraph could use a rewrite as its slightly confusing to read, the intent is somewhat muddied IMHO

**Review by lmbr-pip — APPROVED** [review]

File: Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md [hunk 5]
```diff
@@ -1,35 +1,40 @@
...
-Or follow **AWS CLI** configuration which can be reused by boto3 lib:
+Or follow **AWS CLI** configuration directions which can be reused by the boto3 lib:
  * Follow
    [Quick configuration with aws configure](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-config)
 
 ## Python Environment Setup Options
-### 1. Engine python environment (Including Editor)
-1. In order to use engine python environment, it requires to link Qt binaries for this tool.
+### 1. Use Engine python environment (Including Editor)
+1. In order to use the Open 3D engine's python environment, this tool requires linking the Qt binaries.
```
Comment (AMZN-puvvadar) [inline Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md:17]: NP: I think the engine in Open 3D Engine needs to be capitalized

File: Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md [hunk 6]
```diff
@@ -1,35 +1,40 @@
...
 Follow cmake instructions to configure your project, for example:
    ```
    $ cmake -B <BUILD_FOLDER> -S . -G "Visual Studio 16 2019" -DLY_PROJECTS=<PROJECT_NAME>
    ```
 
-2. At this point, double check engine python environment gets setup under *<ENGINE_ROOT_PATH>/python/runtime* directory
+2. At this point, double check that the Open 3D engine's python environment gets set up under *<ENGINE_ROOT_PATH>/python/runtime* directory
```
Comment (AMZN-puvvadar) [inline Gems/AWSCore/Code/Tools/ResourceMappingTool/README.md:23]: Same here

**Review by AMZN-puvvadar — APPROVED** [review]

**Review by lmbr-pip — APPROVED** [review]
